codex/raw/marketplace.md

```diff
@@ -0,0 +1,264 @@
+
+Client nodes can create storage requests on the Codex network via the Codex marketplace.
+The marketplace is a smart contract deployed on a EVM-compatiable blockchain.
+The smart contract MUST handle storage request, storage slot state, 
+storage provider rewards, storage provider collateral, and storage proof state. 
+
+To create a request to store a dataset on the Codex network,
+client nodes MUST split the dataset into data chunks, $(c_1, c_2, c_3, \ldots, c_{n})$.
+Using a erasure coding technique, 
+the data chunks are encoded and placed into separate slots.
+The erasure coding technique SHOULD be the [Reed-Soloman algorithm](https://hackmd.io/FB58eZQoTNm-dnhu0Y1XnA).
+
+The requester, client node, SHOULD submit a transaction with the desired request parameters.
+The requester MUST provide a `duration` value for the storage request along with the appropriate `reward`,
+payment for request. 
+Once a request is created via the transaction, 
+all slots MUST be filled by storage providers before the request is officially started.
+If the request does not attract enough storage providers after a time set by the requester,
+which is defined by as `expiry`,
+the request MUST be `canceled`.
+If canceled, `collateral` SHOULD be returned to any storage providers and 
+`reward` returned to the requester.
+The requester MAY create a new request with different values to restart the process.
+
+The storage request SHOULD initiate a new storage contract with the following parameters:
+
+```solidity
+
+function requestStorage() public {
+  // the Codex node requesting storage
+  address client;
+
+  // content identifier
+  string cid;
+
+  // merkle root of the dataset, used to verify storage proofs
+  byte32 merkleRoot;
+
+  // tokens from the requester to reward storage providers
+  uint256 reward;
+
+  // amount of tokens required for collateral by storage providers
+  UInt256 collateral;
+
+  // frequency that proofs are checked by validator nodes
+  uint256 proofProbability;
+
+  // amount of desired time for stoage request
+  uint256 duration;
+
+  // the number of requested slots
+  uint64 slots;
+
+  // amount of storage per slot
+  uint256 slotSize;
+
+  // Amount of time before request expires
+  uint256 expiry;
+
+  // random value to differentiate from other requests
+  byte32 nonce;
+
+}
+
+```
+
+`cid` 
+
+An identifier used to locate the dataset
+- MUST be a sha-256 hash (length of 32 bytes), [CIDv1](https://github.com/multiformats/cid#cidv1)
+- MUST be generated by the client node
+
+`reward`
+
+- it MUST be a token known to the network.
+- it MUST be paid directly to storage providers who fill slots for the `duration` of the request
+
+`collateral`
+
+All storage providers MUST provide token collateral before being able to fill a storage slot.
+The following is related to storage provider who has offered `collateral`
+
+If a storage provider, filling a slot,
+fails to provide enough proofs of storage, the `collateral` MUST be forfeited.
```

Comment:
Added this statement, as this could be implemented in different ways. "This MAY be managed by updating a smart contract object that tracks the number of missed proofs, percentage of `collateral` already slashed, or number of slashed `collateral` for slot to be freed. "